### DISCUSSION TITLE
GPG Keys & Configuration

### DISCUSSION THREAD

**Discussion by Apocrathia** [hassio-addons/addon-vscode#495] [Ideas]
I've currently got an issue where I had to do a reinstall of my HAOS, and lost the GPG key that I had put into the add-on. Not a huge deal, I'll generate another key, de-auth the old one, and add it to my GitHub. 

However, it has made me curious about adding GPG to the add-on configuration itself. I'd be happy to take a stab at this if anyone else would be interested. Does this sound like a useful feature for anyone else?

**Comment by frenck** [comment]
> However, it has made me curious about adding GPG to the add-on configuration itself. I'd be happy to take a stab at this if anyone else would be interested. Does this sound like a useful feature for anyone else?

How would that be a benefit over the current method? In both cases they are part of the backup?

**Comment by Apocrathia** [comment]
I think I had written a script to install it and it didn't get backed up. No clue what happened, but it didn't get restored when I last reinstalled the add-on. What would the recommended method be to add a GPG key? I was planning on following GitHub's docs on them as I did before.

https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key

**Reply by frenck** [reply]
Honestly? I would not use GPG for signing as of today, as SSH keys are now supported for signing (GitHub recently released that).

You can use `/data` to store your GPG key.

**Reply by Apocrathia** [reply]
I've tried using my SSH key, but the version of git included in the add-on doesn't support that. 
```
➜  /config git:(master) ✗ git --version
git version 2.30.2
```
Support for SSH key signing was added in 2.34. https://github.blog/2021-11-15-highlights-from-git-2-34/

Unfortunately, the version in the Debian Bullseye repo is still 2.30.
```
➜  /config git:(master) ✗ apt-get upgrade git
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
git is already the newest version (1:2.30.2-1).
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

SSH keys are way less cumbersome to work with, so I completely agree with using SSH > GPG, but it doesn't appear to be possible without manually upgrading git.

**Comment by Apocrathia** [comment]
So, I've got my keys saved in `/data`, but the `~/.gnupg` folder doesn't get linked automatically. I'd like to copy [this block](https://github.com/hassio-addons/addon-vscode/blob/8f29272815f923e8f96c1da54b9bee9070c1cb66/vscode/rootfs/etc/cont-init.d/user.sh#L18) from the `user.sh` script to also perform a `ln -s /data/.gnupg ~/.gnupg` as on [L27](https://github.com/hassio-addons/addon-vscode/blob/8f29272815f923e8f96c1da54b9bee9070c1cb66/vscode/rootfs/etc/cont-init.d/user.sh#L27) in the same script. Would you mind if I submit a PR for that?
